plyer/platforms/linux/filechooser.py

```diff
@@ -68,7 +68,9 @@ def _run_command(self, cmd):
                     self._handle_selection(self.selection)
                     return self.selection
                 else:
-                    return None
+                    self.selection = None
+                    self._handle_selection(self.selection)
+                    return self.selection
```

Comment:
seems like it would be worth extracting that duplicate logic to a helper method. e.g:
```py
def set_and_return_selection(self, value):
    self.selection = value
    self._handle_selection(value)
    return value
```

and use it
```py
    if ret is not None:
        ...
        return self.set_and_return_selection(self._split_output(out)
    else:
        return self.set_and_return_selection(None)
```

Reply:
Done

https://user-images.githubusercontent.com/40869738/160231054-b827071b-af6b-4104-9add-98c1782b13ea.mp4